### ISSUE TITLE
Samples build.zig cannot be directly used

### ISSUE THREAD

**Issue by Pyrolistical** [zig-gamedev/zig-gamedev#597]
All the samples depend on the top level `build.zig` via:
```zig
const Options = @import("../../build.zig").Options;
```

I think we can refactor to directly pass in `target` and `optimize` but duplicate all other options as `b.option`.

**Comment by Pyrolistical** [comment]
I got it working, but apparently `b.option` can only be declared once. So I see why the `Options` struct was created.

You can see the diff https://github.com/Pyrolistical/zig-gamedev/pull/1/files

**Comment by hazeycode** [comment]
Yes the build.zigs for samples were not intended to be run using zig build.

Is there any benefit to doing this over just running a particular sample build command from the top level?

**Comment by Pyrolistical** [comment]
I'm trying to add a new live editing sample, which recompiles and reloads a dynamic library upon save. That requires some build trickery which, is why I'm running into this issue.

**Comment by Pyrolistical** [comment]
Oh I found another way to break the `Options` dep with an `anytype`:
https://github.com/Pyrolistical/zig-gamedev/pull/2/files

**Comment by hazeycode** [comment]
> Oh I found another way to break the `Options` dep with an `anytype`: https://github.com/Pyrolistical/zig-gamedev/pull/2/files

I think we should make this change anyway. Sample apps only consume a self-defined subset of options; and `anytype` is great for implicit interfaces.

**Comment by Pyrolistical** [comment]
I think this is blocked by `samples/common`.

For example for `samples/audio_experiments`:
1. Top-level build: sample uses `@import("../common/build.zig")`. This is allowed since both `samples/audio_experiments` and `samples/common` are children of cwd
2. Direct sample build: Can't use `@import("../common/build.zig")` since cwd is `samples/audio_experiments` and `../common` is not a child of cwd. This could be solved by adding `common` to `build.zig.zon` and `@import("common")`, but then we have `@import("common")` and `@import("../common/build.zig")` in the same file and zig doesn't like that. It sees it as a double import.
